numpy/core/function_base.py

```diff
@@ -408,8 +408,18 @@ def geomspace(start, stop, num=50, endpoint=True, dtype=None, axis=0):
 
     log_start = _nx.log10(start)
     log_stop = _nx.log10(stop)
-    result = out_sign * logspace(log_start, log_stop, num=num,
-                                 endpoint=endpoint, base=10.0, dtype=dtype)
+    result = logspace(log_start, log_stop, num=num,
+                      endpoint=endpoint, base=10.0, dtype=dtype)
+
+    # Make sure the endpoints match the start and stop arguments. This is
+    # necessary because np.exp(np.log(x)) is not necessarily equal to x.
+    if num > 0:
+        result[0] = start
+        if num > 1 and endpoint:
+            result[-1] = stop
```

Comment:
This code isn't correct if `axis != 0` unfortunately

Reply:
Happy to fix if you can tell me what's wrong, because all tests are green, including the ones where `axis != 0`. (There is an Azure pipeline check that fails, but that seems to be unrelated to my change as it's an [HTTP 500 internal server error](https://dev.azure.com/numpy/27346c6a-2774-4eac-bf85-e068127c0ccc/_apis/build/builds/10403/logs/44))

Reply:
Sure, happy to elaborate, was on mobile so I thought I'd leave a breadcrumb in case other reviewers got back to it.

I think the following case fails with this patch:
```
np.geomspace(np.array([1, 2]), np.array([100, 200]), 3, axis=1)
```

Reply:
That results in

```
[[  1.  10. 100.]
 [  2.  20. 200.]]
```
independently of whether my patch is applied or not. Note that the `moveaxis()` call is still there - *after* my code change.

Reply:
Apologies, you're correct - this function doesn't work how I thought it did.